vmdb/spec/models/ems_openstack_infra_spec.rb

```diff
@@ -44,4 +44,36 @@
     end
   end
 
+  context "provider_hooks" do
+    before :each do
+      @ems = FactoryGirl.create(:ems_openstack_infra_with_authentication)
+    end
+
+    it "creates related ProviderOpenstack after creating EmsOpenstackInfra" do
+      @ems.provider.name.should equal @ems.name
+      @ems.provider.zone.should == @ems.zone
+      ProviderOpenstack.count.should equal 1
```

Comment:
it tests the hook, that creating EmsOpenstackInfra creates also ProviderOpenstack